server/src/api.rs

```
@@ -44,3 +50,55 @@ async fn get_user_metadata(
 
     Ok(Json(Ok(Some(meta))))
 }
+
+#[web::delete("/metadata/bulk")]
+async fn delete_metadata_bulk(
+    state: State<AppState>,
+    req: Json<BulkUsers>,
+    http_req: web::HttpRequest,
+) -> Result<Json<ApiResult<()>>> {
+    // verify token
+    let token = http_req
+        .headers()
+        .get("Authorization")
+        .ok_or(Error::AuthToken("missing Authorization header".to_string()))?
+        .to_str()
+        .map_err(|_| Error::AuthToken("invalid Authorization header".to_string()))?;
+    let token = token.trim_start_matches("Bearer ");
+    match verify_token(token) {
+        Ok(true) => (),
+        Ok(false) => return Err(Error::AuthToken("invalid token".to_string())),
+        Err(e) => return Err(Error::AuthToken(e.to_string())),
+    }
+
+    let keys = req.users.iter().map(|p| p.to_text()).collect::<Vec<_>>();
+
+    let mut conn = state.redis.get().await?;
+
+    let futures_iter = keys.iter().map(|key| async {
+        let res = conn
+            .clone()
+            .hdel::<_, _, bool>(key.to_string(), METADATA_FIELD)
+            .await;
+        match res {
+            Ok(_) => Ok(()),
+            Err(e) => Err((key.to_string(), e.to_string())),
+        }
+    });
+    let stream = futures::stream::iter(futures_iter)
+        .boxed()
+        .buffer_unordered(100);
+    let results = stream.collect::<Vec<Result<_, (String, String)>>>().await;
+    let errors = results
+        .into_iter()
+        .filter_map(|res| match res {
+            Ok(_) => None,
+            Err((key, e)) => Some((key, e)),
+        })
+        .collect::<Vec<_>>();
+    if !errors.is_empty() {
```

Return the errors in ApiResult instead of logging.